internal/kafka/internal/services/kafka.go

```diff
@@ -960,52 +962,107 @@ type KafkaStatusCount struct {
 	Count  int
 }
 
-type KafkaRegionCount struct {
+type KafkaStreamingUnitCountPerRegion struct {
 	Region        string
-	InstanceType  string `gorm:"column:instance_type"`
-	ClusterId     string `gorm:"column:cluster_id"`
+	InstanceType  string
+	ClusterId     string
 	Count         float64
-	CloudProvider string `gorm:"column:cloud_provider"`
+	CloudProvider string
 }
 
-func (k *kafkaService) CountByRegionAndInstanceType() ([]KafkaRegionCount, error) {
+func (kafkaStreamingUnitCountPerRegion KafkaStreamingUnitCountPerRegion) isSame(kafkaPerRegionFromDB *KafkaPerRegionCount) bool {
+	return kafkaStreamingUnitCountPerRegion.CloudProvider == kafkaPerRegionFromDB.CloudProvider &&
+		kafkaStreamingUnitCountPerRegion.ClusterId == kafkaPerRegionFromDB.ClusterId &&
+		kafkaStreamingUnitCountPerRegion.InstanceType == kafkaPerRegionFromDB.InstanceType &&
+		kafkaStreamingUnitCountPerRegion.Region == kafkaPerRegionFromDB.Region
+}
+
+// KafkaPerRegionCount is a struct used to query the database using a "group by" clause
+type KafkaPerRegionCount struct {
+	Region        string
+	InstanceType  string
+	ClusterId     string
+	Count         float64
+	CloudProvider string
+	SizeId        string
+}
+
+type ClusterSelection struct {
+	CloudProvider         string
+	ClusterID             string
+	Region                string
+	SupportedInstanceType string
+}
+
+func (k *kafkaService) CountStreamingUnitByRegionAndInstanceType() ([]KafkaStreamingUnitCountPerRegion, error) {
+
+	var clusters []*ClusterSelection
 	dbConn := k.connectionFactory.New()
+	if err := dbConn.Model(&api.Cluster{}).Scan(&clusters).Error; err != nil {
```

Comment:
Will there be a chance where `ClusterID` may be empty when using dynamic scaling? :thinking:

Reply:
That's true, especially if the cluster has just been accepted. 

We could ignore accepted clusters / clusters that do not have the cluster id yet, but I'll prefer to leave it when we've dynamic scaling in full flow. What do you think?

Reply:
thanks and yep I agree :+1: